exercises/src/main/scala/fpinscala/parallelism/Nonblocking.scala

```diff
@@ -55,30 +65,36 @@ object Nonblocking {
     def eval(es: ExecutorService)(r: => Unit): Unit =
       es.submit(new Callable[Unit] { def call = r })
 
-
     def map2[A,B,C](p: Par[A], p2: Par[B])(f: (A,B) => C): Par[C] =
       es => new Future[C] {
-        def apply(cb: C => Unit): Unit = {
+        def apply(cb: C => Unit)(kErr: Throwable => Unit): Unit = {
+          def cbOrErr(a: A, b: B) =
+            try { cb(f(a, b)) }
+            catch { case err: Throwable => kErr(err) }
```

Comment:
@gpoirier yeah true, with this error callback design I couldn't get it to work any other way except with a `Throwable`.